drivers/storage/class/class2/class2.c

```diff
@@ -4052,6 +4052,40 @@ Return Value:
         goto SetStatusAndReturn;
     }
 
+    if (irpStack->Parameters.DeviceIoControl.IoControlCode == IOCTL_MOUNTDEV_QUERY_DEVICE_NAME) {
+
+        PMOUNTDEV_NAME name = Irp->AssociatedIrp.SystemBuffer;
+
+        if (irpStack->Parameters.DeviceIoControl.OutputBufferLength < sizeof(MOUNTDEV_NAME)) {
```

Comment:
Should `FIELD_OFFSET(MOUNTDEV_NAME, Name)` be enough?

Reply:
No. Seen MS example, they except a full structure.